File: 1040.cpp
#include <iostream>
#include <iomanip>

using namespace std;

int main(){
    float nota1, nota2, nota3, nota4, notaExame, media;
    cin >> nota1 >> nota2 >> nota3 >> nota4;
    media = ((nota1 * 2) + (nota2 * 3) + (nota3 * 4) + (nota4 * 1)) / (2 + 3 + 4 + 1);
    cout << fixed << setprecision(1);
    cout << "Media: "<< media << endl;
    if (media >= 7.0){
        cout << "Aluno aprovado." << endl;
    } else if (media >= 5.0 && media <= 6.9){
        cout << "Aluno em exame." << endl;
        cin >> notaExame;
        cout << "Nota do exame: " << notaExame << endl;
        media = (media + notaExame) / 2;
        if (notaExame >= 5.0){
            cout << "Aluno aprovado." << endl;
            cout << "Media final: " << media << endl;
        } else {
            cout << "Aluno reprovado." << endl;
            cout << "Media final: " << media << endl;
        }
    } else {
        cout << "Aluno reprovado." << endl;
    }

    return 0;
}

File: 1040.py
N1, N2, N3, N4 = input().split()
N1, N2, N3, N4 = float(N1), float(N2), float(N3), float(N4)
media = ((N1 * 2) + (N2 * 3) + (N3 * 4) + (N4 * 1)) / 10
if media >= 7.0:
    print('Media: {:.1f}'.format(media))
    print('Aluno aprovado.')
elif media < 5.0:
    print('Media: {:.1f}'.format(media))
    print('Aluno reprovado.')
else:
    print('Media: {:.1f}'.format(media))
    print('Aluno em exame.')
    notaexame = float(input())
    print('Nota do exame: {:.1f}'.format(notaexame))
    mediafinal = (notaexame + media) / 2 
    if mediafinal >= 5.0:
        print('Aluno aprovado.')
        print('Media final: {:.1f}'.format(mediafinal))
    else:
        print('Aluno reprovado.')
        print('Media final: {:.1f}'.format(mediafinal))